src/components/DateInput/DateInput.tsx

```diff
@@ -102,16 +112,20 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
     close();
   }, [close, value, timezone, setCurrentDate]);
 
+  const onClear = useCallback(() => setCurrentDate(undefined), [setCurrentDate]);
+
   const onApply = useCallback(
     e => {
-      if (!currentDate) {
+      if (!currentDate && disableReset) {
```

Comment:
Although I understand that you can't reset if `disableReset` is enabled (thus the `currentDate` will never turn to `undefined`), I still feel that this may lead into some bugs.

This an optional proposal, but what I'd reccommend iss cheecking the previous value of `currentDate` to decide whether you should return early  or not.

- If prev = undefined & current = undefined --> it's the first render
- If prev = something & current = undefined --> it was reset

@vorillaz thoughts on this?

Reply:
an implementation of this would be:

```tsx
function usePrevious<T>(value: T) {
  const ref = useRef<T>();
  
  React.useEffect(() => {
    ref.current = value;
  }, [value]);

  return ref.current;
}
```

and then do:

```tsx
const previousDate = usePrevious(currentDate)
```